ethers-contract/src/call.rs

```diff
@@ -104,6 +104,39 @@ impl<M: Middleware> ContractError<M> {
         }
     }
 
+    /// extract reason msg from a revert error
+    ///
+    /// ## Returns
+    /// `None` if we couldnt decode
+    /// `Some(string)` if we could get the revert message. The string has no null bytes
+    ///
+    /// ## NOTE
+    ///
+    /// A lot of ethers RPC do not enable this by default. Make sure yours has it enabled.
+    /// for more details see https://docs.soliditylang.org/en/v0.8.12/control-structures.html#revert
+    pub fn get_revert_msg(&self) -> Option<String> {
+        self.as_revert().and_then(|data: &Bytes| {
+            let bytes = data.to_vec();
+            let mut bytes = bytes.as_slice();
+            let data_string: &[u8];
+
+            (_, bytes) = bytes.split_at(4); // first 4 bytes for function selector
```

Comment:
Might be worth returning `None` if the selector is not `08c379a0` (`keccak256("Error(string)")`). You can achieve this by doing `bytes[..4] != [0x08, 0xc3, 0x79, 0xa0]`